src/parser/semantic_check.c

```diff
@@ -11424,6 +11424,14 @@ pt_assignment_compatible (PARSER_CONTEXT * parser, PT_NODE * lhs, PT_NODE * rhs)
 		  if (rhs->node_type != PT_HOST_VAR)
 		    {
 		      d = tp_domain_resolve_default (lhs_dbtype);
```

Comment:
If resolving a domain by default is a problem, how about using a function that doesn't resolve by default?

```suggestion
		      d = tp_domain_resolve(lhs_dbtype, NULL, sci.prec, sci.scale, NULL, sci.coll_infer.coll_id);
```

Reply:
Good point, thanks.
`d = tp_domain_resolve (lhs_dbtype, NULL, sci.prec, sci.scale, NULL, 0);`
the collation looks not needed for numeric type.